noelia-melgarejo/lesson1/Auto/src/Auto.java

```diff
@@ -0,0 +1,131 @@
+public class Auto {
+    private double gasCapacity;
+    private double gasConsumption;
+    private double gas;
+    private double mileage;
+    private boolean isPowerOn;
+
+    /**
+     * The constructor method
+     * @param gasConsumption in Litre/KM
+     */
+    public Auto(double gasConsumption) {
+        this.gasConsumption = gasConsumption;
+        mileage = 0;
+        isPowerOn = false;
+        gas = 0;
+
+    }
+    /**
+     * gets amount of Gasoline
+     * @return
+     */
+    public double getGas() {
+        return gas;
+    }
+
+    /**
+     * Sets a amount the gas
+     * @param gas
+     */
+    public void setGas(double gas) {
+        this.gas = gas;
+    }
+
+    /**
+     * Gets the gas Capacity
+     * @return
+     */
+    public double getGasCapacity() {
+        return gasCapacity;
+    }
+
+    /**
+     * Sets of gas capacity
+     * @param gasCapacity
+     */
+    public void setGasCapacity(double gasCapacity) {
+        this.gasCapacity = gasCapacity;
+    }
+
+    /**
+     * Gets the gas Consumption L/Km
+     * @return
+     */
+    public double getGasConsumption() {
+        return gasConsumption;
+    }
+
+    /**
+     * Sets the gasConsumption
+     * @param gasConsumption
+     */
+    public void setGasConsumption(double gasConsumption) {
+        this.gasConsumption = gasConsumption;
+    }
+
+    /**
+     * Gets the mileage
+     * @return
+     */
+    public double getMileage() {
+        return mileage;
+    }
+
+    /**
+     * Sets the mileage
+     * @param mileage
+     */
+    public void setMileage(double mileage) {
+        this.mileage = mileage;
+    }
+
+    /**
+     * Gets true for power on
+     * @return
+     */
+    public boolean powerOn() {
+        return isPowerOn = true;
+    }
+
+    /**
+     * Sets power Off
+     */
+    public void powerOff() {
+        isPowerOn = false;
+    }
+
+    /**
+     * Loads amount of gas
+     * @param gas Liters
+     */
+    public void loadGas(double gas) {
+        if (gasCapacity > gas)
+            setGas(gas);
+        else this.gas = gasCapacity;
+    }
+
+    /**
+     * Moves the car
+     * @param distanceToMove expressed in Km
+     */
+    public void move(double distanceToMove) {
+        mileage = distanceToMove;
```

Comment:
milage should increase not be replace and this should be done after determining the distance. Also you need to use distanceToMove for this calculation, moreover consider that probably you will not be able to move the required distance in this case the car should move the max distance it can with its actual gas